packages/grid/_modules_/grid/hooks/features/rows/useGridRows.ts

```diff
@@ -174,16 +175,20 @@ export const useGridRows = (
           addedRows.push(partialRow);
           return;
         }
-        const lookup = { ...internalRowsState.current.idRowsLookup };
 
-        lookup[id] = {
+        if (!updatedLookup) {
```

Comment:
when can that be true?

Reply:
Only on the very first iteration (we could initialize `updatedLookup` too, but that would waste resource if we only add/remove rows)